tests/test_runs/test_run_functions.py

```diff
@@ -119,14 +119,14 @@ def _wait_for_processed_run(self, run_id, max_waiting_time_seconds):
         # time.time() works in seconds
         start_time = time.time()
         while time.time() - start_time < max_waiting_time_seconds:
-            run = openml.runs.get_run(run_id, ignore_cache=True)
 
             try:
                 openml.runs.get_run_trace(run_id)
             except openml.exceptions.OpenMLServerException:
                 time.sleep(10)
                 continue
 
+            run = openml.runs.get_run(run_id, ignore_cache=True)
```

Comment:
Sometimes test fail below because `run.evaluations` is `None`. But I am not able to reproduce this locally. However, the fails do seem fewer now that I move this line (which makes sense, initially there is a race condition where perhaps the trace isn't yet processed in get `get_run` call, but is by the time the `get_run_trace` is called.